### PR TITLE
Build UI of tip calculator app for -- mobile version 

### PR DESCRIPTION
**Pull request by M-AminAlizadeh**
The things that I did in this PR:

1. Build the initial structure of the app using the `BEM` method
2. Style for mobile version only using vanilla CSS
3. Using CSS variables
4. Using rem instead of px

### CODE REVIEW

File: tip-calculator-app-main/assets/content/toDoList.md
Comment (ahangarha) [inline tip-calculator-app-main/assets/content/toDoList.md:1]: Same comment as above

File: tip-calculator-app-main/index.html [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        </section>
+        
+        <section class="select-tip">
+          <h2 class="select-tip__header">Select Tip %</h2>
+          <div class="select-tip__btns">
+            <button class="select-tip__btn">5%</button>
+            <button class="select-tip__btn">10%</button>
+            <button class="select-tip__btn">15%</button>
+            <button class="select-tip__btn">25%</button>
+            <button class="select-tip__btn">50%</button>
```
Comment (ahangarha) [inline tip-calculator-app-main/index.html:34]: These buttons should change the cursor on hover. However, this is not necessary here if you are aiming for mobile. Later can be implemented.

File: index.html
Comment (ahangarha) [inline index.html:1]: There are two issues:
1. Is this change related to the PR title?
2. The page doesn't seem to be ready for a merge:
    image

I think you want to have a homepage where you can list all the projects. This is good 👍🏾 but do not mix its development with other PRs related to the sub-projects.

File: index.html [hunk 2]
```diff
@@ -4,74 +4,61 @@
...
       type="image/x-icon" />
-    <!-- Bootstrap Css CDN -->
-    <link
-      href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/css/bootstrap.min.css"
-      rel="stylesheet"
-      integrity="sha384-rbsA2VBKQhggwzxH7pPCaAqO46MgnOM80zW1RWuH61DGLwZJEdK2Kadq2F9CUG65"
-      crossorigin="anonymous" />
-    <!-- Title -->
-    <title>Frontend Mentor Projects -- Vanilla Version</title>
+    <link rel="stylesheet" href="./style/output.css" />
```
Comment (ahangarha) [inline index.html:11]: The output.css file should be accessible after a build process and should get optimized for production (by modification for example).
- It shouldn't be committed and pushed.
- Consider adding a build script in `package.json` file.

Also, see other comments related to separating the development of your homepage project from sub-projects. It becomes confusing when they get mixed in one single RP.

File: tip-calculator-app-main/index.html [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      rel="shortcut icon"
+      href="https://img.icons8.com/color/48/charity-donation.png"
+      type="image/x-icon" />
+    <title>Tip Calculator App</title>
+  </head>
+  <body>
+    <main class="container">
+      <div class="brand-container">
+        <img src="./assets/images/logo.svg" alt="logo" />
+      </div>
```
Comment (ahangarha) [inline tip-calculator-app-main/index.html:17]: Maybe better to use `section` here to follow the same pattern of having several sections withing `main` tag.

File: tip-calculator-app-main/index.html [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      <section class="tip-calculator">
+        <section class="bill">
+          <h2 class="bill__header">Bill</h2>
+          <div class="bill__input-container">
+            <img src="./assets/images/icon-dollar.svg" alt="dollar icon" />
```
Comment (ahangarha) [inline tip-calculator-app-main/index.html:22]: What if using a simple `$` text rather than making a call to get an SVG file containing `$` letter? Besides saving one unnecessary call, we get the chance to make it follow the same typography as the app.

File: tip-calculator-app-main/style/style.css [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+  color: var(--natural-cyan-dark);
+}
+
+.select-tip__input {
+  background-color: var(--natural-grayish-cyan-very-light);
+  width: 100%;
+  border: none;
+  font-size: 1.5rem;
+  outline: none;
+  direction: rtl;
```
Comment (ahangarha) [inline tip-calculator-app-main/style/style.css:110]: Why `rtl`?

File: tip-calculator-app-main/style/style.css [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+.bill__input,
+.number-of-people__input {
+  width: 95%;
+  border: none;
+  background-color: var(--natural-grayish-cyan-very-light);
+  color: var(--natural-cyan-dark);
+  font-size: 1.5rem;
+  direction: rtl;
```
Comment (ahangarha) [inline tip-calculator-app-main/style/style.css:77]: Why `rtl`?

File: tip-calculator-app-main/index.html [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+          </div>
+          <div class="show-screen__total-container">
+            <h3 class="show-screen__total__header">
+              Total
+              <p class="show-screen__total__header__person">/ person</p>
+            </h3>
+            <div class="show-screen__total__value">$32.79</div>
+          </div>
+          
+          <button class="show-screen__reset-btn">Reset</button>
```
Comment (ahangarha) [inline tip-calculator-app-main/index.html:67]: This button doesn't have any hover effect.

File: .prettierignore [hunk 8]
```diff
@@ -1,2 +1,3 @@
-index.html
-Results-summary-component/
\ No newline at end of file
+./index.html
+Results-summary-component/
+# tip-calculator-app-main/
```
Comment (ahangarha) [inline .prettierignore:3]: Why ignore these files and projects?

File: tip-calculator-app-main/index.html (hunk 1 above)
Comment (M-AminAlizadeh) [reply tip-calculator-app-main/index.html:34]: Sure I will add the hover effect for the desktop version

File: index.html
Comment (M-AminAlizadeh) [reply index.html:1]: Your right 
I checked the git log strictly to find the issue in my git flow
and I found it

File: tip-calculator-app-main/style/style.css (hunk 5 above)
Comment (M-AminAlizadeh) [reply tip-calculator-app-main/style/style.css:110]: Because of the design
I could use `text-align: right` instead of direction
I think it's better

File: tip-calculator-app-main/index.html (hunk 7 above)
Comment (M-AminAlizadeh) [reply tip-calculator-app-main/index.html:67]: Not for the mobile version 
But I will add it to the desktop version

File: tip-calculator-app-main/index.html (hunk 4 above)
Comment (M-AminAlizadeh) [reply tip-calculator-app-main/index.html:22]: @ahangarha 
It's doable but it needs CSS code
Is writing CSS better than calling a prepared file?

Comment (ahangarha) [reply tip-calculator-app-main/index.html:22]: It is a good question. To me, if something can be done easily with CSS, it is better to do so. It will follow the design with ease.
For complex things, we can do a trade-off.

File: tip-calculator-app-main/style/style.css (hunk 5 above)
Comment (ahangarha) [reply tip-calculator-app-main/style/style.css]: Mind the difference between text alignment and text direction. How did I notice? I put a negative number!

File: .prettierignore (hunk 8 above)
Comment (ahangarha) [reply .prettierignore:3]: Config your prettier to follow the linter.

File: tip-calculator-app-main/index.html (hunk 4 above)
Comment (M-AminAlizadeh) [reply tip-calculator-app-main/index.html:22]: @ahangarha Then I will change it to text and CSS instead of importing image

Comment (ahangarha) [reply tip-calculator-app-main/index.html:22]: It's okay for now. If you choose to do this way, it is okay. If it was my project, I would ask you to change to text (unless it be one of the many icons we use in a given component)
